pennylane/labs/resource_estimation/resource_container.py

```diff
@@ -59,10 +59,10 @@ def __init__(self, op_type, params: dict) -> None:
         if not issubclass(op_type, ResourceOperator):
             raise TypeError(f"op_type must be a subclass of ResourceOperator. Got {op_type}.")
 
-        self._name = (op_type.__name__).replace("Resource", "")
         self.op_type = op_type
         self.params = params
-        self._hashable_params = tuple(params.items())
+        self._hashable_params = _make_hashable(params)
+        self._name = name or op_type.tracking_name(**params)
```

Comment:
Could you add tests for both of these. Specifically, a test which checks that if a custom name is passed it is correctly set and a test which checks that if a dictionary is passed as a parameter then it gets properly converted to a hashable type.